src/components/DataDownloadModal/index.jsx

```diff
@@ -0,0 +1,228 @@
+    const DatasetCheckboxRows = [];
+    Object.keys(dataListCheckBoxes)
+      .forEach((modelName) => {
+        const filesize = parseFloat(
+          `${sampledataRegistry.Models[modelName].filesize / 1000000}`
+        ).toFixed(2) + ' MB';
+        const labelSuffix = sampledataRegistry.Models[modelName].labelSuffix || '';
+        DatasetCheckboxRows.push(
+          <tr key={modelName}>
+            <td>
+              <Form.Check
+                className="pt-1"
+                id={modelName}
+              >
+                <Form.Check.Input
+                  type="checkbox"
+                  checked={dataListCheckBoxes[modelName]}
+                  onChange={(event) => this.handleCheckList(
+                    event, modelName
+                  )}
+                />
+                <Form.Check.Label>
+                  {modelName}
+                </Form.Check.Label>
+              </Form.Check>
+            </td>
+            <td><em>{labelSuffix}</em></td>
+            <td>{filesize}</td>
+          </tr>
+        );
+      });
+
+    return (
+      <Modal className="download-data-modal"
+        show={this.props.show}
+        onHide={this.props.closeModal}
+        size="lg"
+      >
+        <Form>
+          <Modal.Header>
+            <Modal.Title>Download InVEST sample data</Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <Table
+              size="sm"
+              borderless
+              striped
+            >
+              <thead>
+                <tr>
+                  <th>
+                    <Form.Check
+                      type="checkbox"
+                      id="all-sampledata"
+                      checked={this.state.allDataCheck}
+                      onChange={this.handleCheckAll}
+                      name="all-sampledata"
+                      label="Select All"
+                    />
+                  </th>
+                </tr>
+              </thead>
+              <tbody className="table-body">
+                {DatasetCheckboxRows}
+              </tbody>
+            </Table>
+          </Modal.Body>
+          <Modal.Footer>
+            <Button
+              variant="secondary"
+              onClick={this.props.closeModal}
+            >
+              Cancel
+            </Button>
+            <Button
+              variant="primary"
+              onClick={this.handleSubmit}
+              disabled={!downloadEnabled}
+            >
+              Download
+            </Button>
+          </Modal.Footer>
+        </Form>
+      </Modal>
+    );
+  }
+}
+
+DataDownloadModal.propTypes = {
+  show: PropTypes.bool.isRequired,
+  closeModal: PropTypes.func.isRequired,
+  storeDownloadDir: PropTypes.func.isRequired,
+};
+
+export function DownloadProgressBar(props) {
+  const [nComplete, nTotal] = props.downloadedNofN;
+  if (nComplete === nTotal) {
+    return (
+      <Expire delay={props.expireAfter}>
```

Comment:
I'm a little confused about what the Expire component is doing. Is it easy to summarize?